src/charts/line/line-chart.directive.js

```diff
@@ -0,0 +1,287 @@
+                  break;
+                default:
+                  tipRows =
+                    '<tr>' +
+                    '  <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+                    '</tr>';
+                }
+              }
+              return $scope.getTooltipTableHTML(tipRows);
+            },
+            position: function (data, width, height, element) {
+              var center;
+              var top;
+              var chartBox;
+              var graphOffsetX;
+              var x;
+
+              try {
+                center = parseInt(element.getAttribute('x'));
+                top = parseInt(element.getAttribute('y'));
+                chartBox = document.querySelector('#' + $scope.lineChartId).getBoundingClientRect();
+                graphOffsetX = document.querySelector('#' + $scope.lineChartId + ' g.c3-axis-y').getBoundingClientRect().right;
+                x = Math.max(0, center + graphOffsetX - chartBox.left - Math.floor(width / 2));
+
+                return {
+                  top: top - height,
+                  left: Math.min(x, chartBox.width - width)
+                };
+              } catch (e) {
+              }
+            }
+          };
+        };
+
+        /*
+         * Setup Axis options. Default is to not show either axis. This can be overridden in two ways:
+         *   1) in the config, setting showAxis to true will show both axes
+         *   2) in the attributes showXAxis and showYAxis will override the config if set
+         *
+         * By default only line and the tick marks are shown, no labels. This is a line and should be used
+         * only to show a brief idea of trending. This can be overridden by setting the config.axis options per C3
+         */
+
+        if ($scope.showXAxis === undefined) {
+          $scope.showXAxis = ($scope.config.showAxis !== undefined) && $scope.config.showAxis;
+        }
+
+        if ($scope.showYAxis === undefined) {
+          $scope.showYAxis = ($scope.config.showAxis !== undefined) && $scope.config.showAxis;
+        }
+
+        $scope.defaultConfig = $().c3ChartDefaults().getDefaultLineConfig();
+        $scope.defaultConfig.axis = {
+          x: {
+            show: $scope.showXAxis === true,
+            type: 'timeseries',
+            tick: {
+              format: function () {
+                return '';
+              }
+            }
+          },
+          y: {
+            show: $scope.showYAxis === true,
+            tick: {
+              format: function () {
+                return '';
+              }
+            }
+          }
+        };
+
+        // Setup the default configuration
+        $scope.defaultConfig.point.r = 1;
+        $scope.defaultConfig.grid.y.show = false;
+        $scope.defaultConfig.color.pattern = ["#92D400", "#CC0000", "#0088CE", "#9F7FC3"];
+        $scope.defaultConfig.tooltip = $scope.lineTooltip();
+        $scope.defaultConfig.units = '';
```

Comment:
This should be using the default line config provided by Patternfly.
We need to first update c3-charts-defaults to include the new patternfly default chart configs for line chart such as:

      getDefaultLineAxis: patternflyDefaults.getDefaultLineAxis,
      getDefaultLineGrid: patternflyDefaults.getDefaultLineGrid,
      getDefaultLineLegend: patternflyDefaults.getDefaultLineLegend,
      getDefaultLinePoint: patternflyDefaults.getDefaultLinePoint,
      getDefaultLineConfig: patternflyDefaults.getDefaultLineConfig,

The use the new c3-charts-defaults.getDefaultLineConfig here rather than creating the default config.

Note that this config is for multiple lines only and for a single line chart it should use the getDefaultSingleLineConfig Patternfly function (basically just the tooltip changes).